### PR TITLE
Feature/teacher

### LINKED ISSUE #42: Implement React Router
- [ ] Implement React Router

### LINKED ISSUE #58: Create a component for adding a student viewing roster
- [ ] Create a component for adding a student viewing roster

### LINKED ISSUE #59: Create a component to view lessons
- [ ] Create a component to view lessons

### PR DESCRIPTION
**Pull request by KyleWong2510**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- No QuestionCard component

## What is the new behavior?

- Style QuestionCard
- Disable buttons if inputs are empty
- Routing for TeacherHeader buttons
- Add TeacherLesson and TeacherRoster components
- Unit tests for QuestionFormCard
## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

closes #42 
closes #58
closes #59

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
 import './App.scss';
 import CreateLesson from '../CreateLesson/CreateLesson'
+
 function App() {
   return (
     <main className="App">
-      {/* <TeacherHeader />
-      {/* <TeacherLogin /> */}
-      {/* <CreateLesson /> */}
+      <Switch>
```
Comment (Billwilke42) [inline src/App/App.js:15]: Nice work with routing!! Now to get that link generated

File: src/CreateLesson/CreateLesson.js [hunk 2]
```diff
@@ -8,24 +8,28 @@ const CreateLesson = () => {
...
-        setQuestions={setQuestions}
-      />
-      <LessonPreview 
-        lessonTitleText={lessonTitleText} 
-        setLessonTitleText={setLessonTitleText}
-        questions={questions}
-        setQuestions={setQuestions}
-        deleteQuestion={deleteQuestion}
-      />
+      <h1 className='page-title'>Create a Lesson</h1>
```
Comment (Billwilke42) [inline src/CreateLesson/CreateLesson.js:17]: nice work passing these down, we will set it to the store on submit

File: src/App/App.js [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+        <Route path='/login'>
+          <TeacherHeader />
+          <TeacherLogin />
+        </Route>
```
Comment (Melizzo) [inline src/App/App.js:19]: Thank you for setting up the Switch paths!

File: src/QuestionFormCard/AnswerInput/AnswerInput.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, fireEvent } from '@testing-library/react';
+import AnswerInput from './AnswerInput';
+
+describe('AnswerInput', () => {
```
Comment (Billwilke42) [inline src/QuestionFormCard/AnswerInput/AnswerInput.test.js:6]: Nice work on these tests

File: src/QuestionFormCard/QuestionFormCard.js [hunk 5]
```diff
@@ -56,29 +61,52 @@ const QuestionFormCard = ({lessonTitleText, setLessonTitleText, questions, setQu
...
+        />
       </section>
       <section className='question-form-input'>
-        <label htmlFor='answer-input'>Incorrect Answer:</label>
-        <AnswerInput correct={false} answerText={incorrectAnswerText1} setAnswerText={setIncorrectAnswerText1}/>
+        <label htmlFor='answer-input'>Incorrect Answer 1:</label>
+        <AnswerInput 
+          testId='incorrect1'
+          correct={false} 
+          answerText={incorrectAnswerText1}
```
Comment (Melizzo) [inline src/QuestionFormCard/QuestionFormCard.js:87]: This was a great setup for testing the individual incorrect answers

**Review by Billwilke42 — COMMENTED** [review]
Great job! looking forward to getting these endpoints working

**Review by Melizzo — APPROVED** [review]

**Pull request merged**